lib/puppet/daemon.rb

```diff
@@ -165,10 +151,11 @@ def remove_pidfile
     @pidfile.unlock
   end
 
+  # Loop forever running events - or, at least, until we exit.
   def run_event_loop
     agent_run = Puppet::Scheduler.create_job(Puppet[:runinterval], Puppet[:splay], Puppet[:splaylimit]) do
       # Splay for the daemon is handled in the scheduler
-      agent.run(:splay => false)
+      @agent.run(:splay => false)
```

Comment:
since we have an `attr_reader` for agent we can keep the same `agent.run({:splay => false}`